### PR TITLE
Extensions assignment

### PR DESCRIPTION
**Pull request by gilRoz23**
Lior's assignment- not done completly yet

### CODE REVIEW

**Comment by liorsiv973** [comment]
please ask for review again when done :)

File: src/main/kotlin/com/hibob/kotlinEx/extension.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.hibob.kotlinEx
+
+//change the 'sum\ function so that it was declared as an extension to List<Int>
+fun sum(list: List<Int>): Int{
```
Comment (liorsiv973) [inline src/main/kotlin/com/hibob/kotlinEx/extension.kt:4]: remove the Q and leave just the answers

File: src/main/kotlin/com/hibob/kotlinEx/extension.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    var result = 0
+    for (i in list){
+        result +=i
+    }
+    return result
+}
+
+fun List<Int>.sum(): Int {
+    var result = 0
+    for (i in this) {
```
Comment (liorsiv973) [inline src/main/kotlin/com/hibob/kotlinEx/extension.kt:14]: great! just for future uses please look at map{}

File: src/main/kotlin/com/hibob/kotlinEx/extension.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        result += i
+    }
+    return result
+}
+
+//write the following program method as an infix extension function
```
Comment (liorsiv973) [inline src/main/kotlin/com/hibob/kotlinEx/extension.kt:20]: where is the answer?

File: src/main/kotlin/com/hibob/kotlinEx/extension.kt (hunk 2 above)
Comment (gilRoz23) [reply src/main/kotlin/com/hibob/kotlinEx/extension.kt:6]: couldnt figure how to use the map so coded it diffrentially
I'll be glad if you'll show me how you would do it

**Review by liorsiv973 — APPROVED** [review]
hope you tried the map function in diff ex. please fix the name of the pr

**Pull request merged**